### ISSUE TITLE
Don't flag valid identifiers as misspelled in doc comments

### ISSUE THREAD

**Issue by sdankel** [drahnr/cargo-spellcheck#329]
**Is your feature request related to a particular use-case?**

I tried this out, hoping to fix a bunch of typos in doc comments in my repo. However, there are thousands of errors everywhere that I have a comment with an valid identifier, for example:

```
/// A [HashMap] that can hold multiple values and ...
```

![image](https://github.com/drahnr/cargo-spellcheck/assets/47993817/e3928d7e-5913-4577-bb49-409f2758689d)

A possible workaround for now is using the regex transform to ignore anything inside of square brackets, but I haven't been able to get it to work. I think it should be a built-in option since it is conventional to use square brackets around types in rust doc comments for features like hover documentation.

**Describe the solution you'd like to implement/see implemented**

Ideally I wouldn't have to ignore everything inside of square brackets, as the parser could tell if the identifier is a valid name or not.

If you have any suggestions about the regex transform, it would be much appreciated!

**Describe alternatives you've considered**

Use a different spell checker

**Comment by drahnr** [comment]
Hey, thank you for flagging. Commonly I'd expect any inline code or types to be enclosed in back ticks, this includes linkage i.e.

```raw
He [`HashMap`] ghji.
```
